addons/point_of_sale/models/pos_order.py

```diff
@@ -952,6 +1005,38 @@ def _launch_stock_rule_from_pos_order_lines(self):
                 pickings_to_confirm.action_confirm()
         return True
 
+    def _is_product_storable_fifo_avco(self):
+        self.ensure_one()
+        return self.product_id.type == 'product' and self.product_id.cost_method in ['fifo', 'average']
+
+    def _compute_total_cost(self, stock_moves):
+        """
+        Compute the total cost of the order lines.
+        :param stock_moves: recordset of `stock.move`, used for fifo/avco lines
+        """
+        for line in self.filtered(lambda l: not l.is_total_cost_computed):
```

Comment:
@trj-odoo you can remove the if line.product_id, because product_id is required on the line so it'll never happen

Reply:
:+1: